pkg/kubelet/kubelet_pods.go

```diff
@@ -1635,7 +1655,12 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 				}
 				// Secondary IP is not set #105320
 				if len(hostIPs) == 2 && len(s.PodIPs) == 1 {
-					s.PodIPs = append(s.PodIPs, v1.PodIP{IP: hostIPs[1].String()})
+					ok, err := utilnet.IsDualStackIPStrings([]string{s.PodIPs[0].IP, hostIPs[1].String()})
```

Comment:
again, I think using `IPFamilyOf != IPFamilyOf` would be clearer here, and get rid of the unnecessary error check

Reply:
done